### PR TITLE
Setup AWS CDK

### PR DESCRIPTION
**Pull request by odrusso**

### CODE REVIEW

File: infra/lib/eduace-backend.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import {CfnOutput, Construct} from "@aws-cdk/core";
+import {LambdaRestApi} from "@aws-cdk/aws-apigateway";
+import {Code, Function as LambdaFunction} from "@aws-cdk/aws-lambda";
+import {Runtime} from "@aws-cdk/aws-lambda";
+
+export class EduaceBackend extends Construct {
+    constructor(scope: Construct, id: string) {
+        super(scope, id);
+
+        const sourceCodeBucket = new Bucket(this, "EduaceBackendLambdaStore")
```
Comment (ashtonmoomoo) [inline infra/lib/eduace-backend.ts:11]: How exactly does this get the python code?

Comment (ashtonmoomoo) [reply infra/lib/eduace-backend.ts:11]: Oh it doesn't, it's just literally an empty bucket with the name "EduaceBackendLambdaStore"

File: infra/lib/eduace-backend.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        })
+
+        const lambdaFunction = new LambdaFunction(this, "EduaceBackendLambdaFunction", {
+            runtime: Runtime.PYTHON_3_9,
+            // Code may not exist at infra build time
+            code: Code.fromBucket(sourceCodeBucket, "build.zip"),
+            // Name of the method that we call in the Python code
+            handler: "main", // TODO: Check this in the code!
+            environment: {
+                FLASK_DEBUG: "false"
```
Comment (ashtonmoomoo) [inline infra/lib/eduace-backend.ts:25]: icr off the top of my head but we may need to set `FLASK_ENV` here too

File: infra/cdk.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "@aws-cdk/core:stackRelativeExports": true,
+    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true,
+    "@aws-cdk/aws-secretsmanager:parseOwnedSecretName": true,
+    "@aws-cdk/aws-kms:defaultKeyPolicies": true,
+    "@aws-cdk/aws-s3:grantWriteWithoutAcl": true,
+    "@aws-cdk/aws-ecs-patterns:removeDefaultDesiredCount": true,
+    "@aws-cdk/aws-rds:lowercaseDbIdentifier": true,
+    "@aws-cdk/aws-efs:defaultEncryptionAtRest": true,
+    "@aws-cdk/aws-lambda:recognizeVersionProps": true,
+    "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021": true
```
Comment (ashtonmoomoo) [inline infra/cdk.json:16]: boilerplate?

Comment (odrusso) [reply infra/cdk.json:16]: Yeah g

**Pull request merged**